source/MRMesh/MRMeshDistance.cpp

```diff
@@ -263,6 +263,31 @@ MeshSignedDistanceResult findSignedDistance( const MeshPart & a, const MeshPart
     return (signedRes.signedDist > 0.0f) ? MeshSignedDistanceResult{res.a, res.b, 0.0f} : signedRes;
 }
 
+MRMESH_API float findMaxDistanceSqOneWay( const MeshPart& a, const MeshPart& b, const AffineXf3f* rigidB2A, float maxDistanceSq )
+{
+    MR_TIMER;
+
+    const auto& meshVerts = b.mesh.points;
+    auto vertBitSet = getIncidentVerts( b.mesh.topology, b.mesh.topology.getFaceIds( b.region ) );
+    if ( !vertBitSet.any() )
+        return 0.0f;
+
+    std::vector<float> distances( vertBitSet.find_last() + 1 );
+
+    BitSetParallelFor( vertBitSet, [&] ( VertId i )
+    {
+        distances[i] = findProjection( meshVerts[i], a, maxDistanceSq, rigidB2A ).distSq;
+    });
+
+    return  *std::max_element( distances.begin(), distances.end() );
```

Comment:
Performance-wise it is not the best solution. Better would be to use `parallel_reduce`, see `Mesh::computeBoundingBox`